src/main.rs

```diff
@@ -438,14 +450,24 @@ impl From<JsonRow> for Vec<String> {
 // This is intended as a low-level function that abstracts over the SQL engine,
 // and whatever result types it returns.
 // Since it uses a vector, statements should be limited to a sane number of rows.
-pub async fn query(connection: &DbConnection, statement: &str) -> Result<Vec<JsonRow>> {
+pub async fn query(
+    connection: &DbConnection,
+    statement: &str,
+    params: &Vec<JsonValue>,
```

Comment:
Let's try making this generic over `Serializable`, and see if that results in simpler code when calling this function.